src/App.jsx

```diff
@@ -14,45 +15,102 @@ export const goodsFromServer = [
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_FIELD_ALPHABETICALLY = 'alphabetically';
+const SORT_FIELD_LENGTH = 'length';
+const SORT_FIELD_REVERSE = 'reverse';
+
+function getPreparedGoods(goods, { sortField, reversedField }) {
+  const preparedGoods = [...goods];
+
+  if (sortField) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SORT_FIELD_ALPHABETICALLY:
+          return good1.localeCompare(good2);
+
+        case SORT_FIELD_LENGTH:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reversedField) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+export const App = () => {
+  const [reversedField, setReversedField] = useState(false);
+  const [sortField, setSortField] = useState('');
+  const visibleGoods = getPreparedGoods(
+    goodsFromServer,
+    { sortField, reversedField },
+  );
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={`button is-info
+          ${
+            sortField !== SORT_FIELD_ALPHABETICALLY
+              ? 'is-light'
+              : ''
+          }`}
+          onClick={() => setSortField(SORT_FIELD_ALPHABETICALLY)}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          type="button"
+          className={`button is-success
+          ${
+            sortField !== SORT_FIELD_LENGTH
+              ? 'is-light'
+              : ''
+          }`}
+          onClick={() => setSortField(SORT_FIELD_LENGTH)}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          className={`button is-warning
+          ${
+            sortField !== SORT_FIELD_REVERSE
+              ? 'is-light'
+              : ''
+          }`}
+          onClick={() => setReversedField(!reversedField)}
+        >
+          Reverse
+        </button>
+
+        <button
+          type="button"
+          className="button is-danger is-light"
+          onClick={() => {
+            setSortField('');
+            setReversedField(false);
+          }}
```

Comment:
```suggestion
          onClick={reset}
```
Consider moving this logic from jsx to a separate function